spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestSnapshotSelection.java

```diff
@@ -44,57 +48,58 @@
 import org.apache.spark.sql.Row;
 import org.apache.spark.sql.RowFactory;
 import org.apache.spark.sql.SparkSession;
-import org.assertj.core.api.Assertions;
-import org.junit.AfterClass;
-import org.junit.Assert;
-import org.junit.BeforeClass;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-@RunWith(Parameterized.class)
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+@ExtendWith(ParameterizedTestExtension.class)
 public class TestSnapshotSelection {
 
-  @Parameterized.Parameters(name = "planningMode = {0}")
+  @Parameters(name = "properties = {0}")
   public static Object[] parameters() {
-    return new Object[] {LOCAL, DISTRIBUTED};
+    return new Object[][] {
+      {
+        ImmutableMap.of(
+            TableProperties.DATA_PLANNING_MODE, LOCAL.modeName(),
+            TableProperties.DELETE_PLANNING_MODE, LOCAL.modeName())
+      },
+      {
+        ImmutableMap.of(
+            TableProperties.DATA_PLANNING_MODE, DISTRIBUTED.modeName(),
+            TableProperties.DELETE_PLANNING_MODE, DISTRIBUTED.modeName())
+      }
+    };
   }
 
   private static final Configuration CONF = new Configuration();
   private static final Schema SCHEMA =
       new Schema(
           optional(1, "id", Types.IntegerType.get()), optional(2, "data", Types.StringType.get()));
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir private Path temp;
 
   private static SparkSession spark = null;
 
-  private final Map<String, String> properties;
-
-  public TestSnapshotSelection(PlanningMode planningMode) {
-    this.properties =
-        ImmutableMap.of(
-            TableProperties.DATA_PLANNING_MODE, planningMode.modeName(),
-            TableProperties.DELETE_PLANNING_MODE, planningMode.modeName());
-  }
+  @Parameter(index = 0)
+  private Map<String, String> properties;
 
-  @BeforeClass
-  public static void startSpark() {
+  @BeforeEach
+  public void startSpark() {
```

Comment:
previously this was started/stopped at the class-level. Any particular reason why we start/stop this now at the test method level?

Reply:
no reason, changed it to `BeforeAll` and `AfterAll`